include/zephyr/drivers/gpio.h

```diff
@@ -752,6 +756,57 @@ static inline int gpio_pin_configure_dt(const struct gpio_dt_spec *spec,
 				  spec->dt_flags | extra_flags);
 }
 
+#ifdef CONFIG_GPIO_GET_CONFIG
```

Comment:
The `__syscall` function should be available regardless of `CONFIG_GPIO_GET_CONFIG` - only the verification and implementation functions should be conditional (see https://docs.zephyrproject.org/latest/kernel/usermode/syscalls.html#implementation-details).

Reply:
The `__syscall`function will not be included if not called. If it is called, but the the verification method and implementation methods are #ifdef'ed out, it will simply call the ‘unimplemented system call’ verifier, so that will not waste memory either. It's all in the documentation, I linked above.

Reply:
What I mean is that if you try to call the syscall function, with userspace enabled and this feature disabled, the compiler complains about the undefined reference to z_impl_* function.
I have checked this trying to compile the hello_world_user example for v2m_beetle and qemu_riscv32.

Reply:
You need to make the z_vrfy_* function conditional as well.

Reply:
It is conditional, I have updated earlier the PR with code that I used to test that behavior.
I've also updated the SDK to zephyr-sdk-0.14.1.

Reply:
Well, then something is wrong. The conditional `z_vrfy_*` and `z_impl_*` functions is how conditional APIs are done in other driver classes (see e.g. conditional CAN-FD syscalls in `zephyr/drivers/can.h`). I have just verified that calling an unimplemented CAN syscall does not lead to a compile-time error.